Static holder objects

Silence-112 opened the pull request:
## What type of PR is this?
1. [ ] Bug
2. [x] Change
3. [x] Enhancement
4. [ ] Miscellaneous

### What have you changed and why?
Information:

Added a "static holder" object for use in map ports, so statics and such aren't bound to a certain building that may or may not exist in that map

### Please specify which Issue this PR Resolves (If Applicable).
"This PR closes #XXXX!"

### Please verify the following.

1. [ ] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?

1. [ ] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps:

********************************************************
Notes:

Silence-112 committed:
Static holder objects

Silence-112 committed:
Remove backup file

Silence-112 committed:
add static holder to milBuildings

Silence-112 committed:
Merge branch 'unstable' into static-holders

stutpip123 committed:
Update fn_initVarCommon.sqf

Silence-112 committed:
Fix: Static holder scope

stutpip123 committed:
Update fn_milBuildings.sqf

Silence-112 committed:
Add: medium/large static holders

Silence-112 committed:
Merge branch 'unstable' into static-holders

Silence-112 committed:
Fix: Merge conflicts

wersal454 commented on A3A/addons/ultimate/CfgVehicles.hpp:
```diff
@@ -1,4 +1,52 @@
+
+class Land_HelipadCircle_F;
+class Land_HelipadSquare_F;
+class A3AU_RebHelipad_Circle_F: Land_HelipadCircle_F {};
```
```suggestion

```

wersal454 commented on A3A/addons/ultimate/CfgVehicles.hpp:
```diff
@@ -1,4 +1,52 @@
+class Land_HelipadCircle_F;
+class Land_HelipadSquare_F;
+class A3AU_RebHelipad_Circle_F: Land_HelipadCircle_F {};
+class A3AU_RebHelipad_Square_F: Land_HelipadSquare_F {};
```
```suggestion

```

wersal454 commented:
<img width="2560" height="1405" alt="image" src="https://github.com/user-attachments/assets/343147a2-3f7f-43ed-9b4f-06d984d06da0" />
<img width="2560" height="1405" alt="image" src="https://github.com/user-attachments/assets/cb830e71-fbde-4ce5-ba60-6ff8d147a869" />

If this is a an intended effect, then it works

I would say this is lgtm

wersal454 commented on A3A/addons/ultimate/CfgVehicles.hpp:
```diff
@@ -1,4 +1,52 @@
+    displayName = "Static Holder (Medium, AT)";
+};
+
+class A3U_StaticHolderMediumAA: A3U_StaticHolderSmall
```
couldn't you inherit this one from A3U_StaticHolderMediumAT?

wersal454 commented on A3A/addons/ultimate/CfgVehicles.hpp:
```diff
@@ -1,4 +1,52 @@
+    displayName = "Static Holder (Large, AT)";
+};
+
+class A3U_StaticHolderLargeAA: A3U_StaticHolderMediumAA
```
same as medium one

Silence-112 replied:
Probably but then the inheritance gets spaghetti

wersal454 replied:
Since you are not changing anything from A3U_StaticHolderLargeAT beside name, you might as well inherit from that

wersal454 commented:
As an idea, placement markers for SAM and radar like in #735 . Though, since those can vary in sizes drastically it might get out of hand. THOUGH, a way to generally dictate where they should be placed would be nice.

Or, instead of adding more markers, simply use large static holders and do a check if they are inside SAM marker space. But then you would need to somehow diversify which holder goes to SAM and which goes to radar.

Silence-112 committed:
Merge branch 'unstable' into static-holders

jwoodruff40 commented:
> As an idea, placement markers for SAM and radar like in #735 . Though, since those can vary in sizes drastically it might get out of hand. THOUGH, a way to generally dictate where they should be placed would be nice.
> 
> Or, instead of adding more markers, simply use large static holders and do a check if they are inside SAM marker space. But then you would need to somehow diversify which holder goes to SAM and which goes to radar.

Sam marker already exists, and iirc radar uses a sam marker as well.

wersal454 commented:
> > As an idea, placement markers for SAM and radar like in #735 . Though, since those can vary in sizes drastically it might get out of hand. THOUGH, a way to generally dictate where they should be placed would be nice.
> > Or, instead of adding more markers, simply use large static holders and do a check if they are inside SAM marker space. But then you would need to somehow diversify which holder goes to SAM and which goes to radar.
> 
> Sam marker already exists, and iirc radar uses a sam marker as well.

Not what I meant. I know sam marker exists. What I'm saying, is that said marker is used as a zone, in which both SAM and radar are both placed semi-randomly (iIrc), so maybe it would be nice to prevent this semi-random behaviour and dictate more exact positions where they should be placed.

Like, on ``this`` base it would be better if sam was on the **left** and radar on the **right**,but on the ``other`` base, it would be better if it was the **opposite**.

UnseenKill committed:
Apply suggestion from @wersal454

Wrong forward declaration; base class is `A3AU_RebHelipad_base_F`. Also, twice the same class defined makes Arma croak.

Co-authored-by: wersal <wroma@bk.ru>
Signed-off-by: UnseenKill <post@oliver-schieche.de>

UnseenKill committed:
Apply suggestion from @wersal454

Wrong forward declaration; base class is `A3AU_RebHelipad_base_F`. Also, twice the same class defined makes Arma croak.

Co-authored-by: wersal <wroma@bk.ru>
Signed-off-by: UnseenKill <post@oliver-schieche.de>

UnseenKill commented on A3A/addons/ultimate/CfgVehicles.hpp:
```diff
@@ -1,4 +1,50 @@
+#define SCOPESTATICS scope = 0; scopeCurator = 0
+
+class Land_HelipadCircle_F;
+class Land_HelipadSquare_F;
```
Unused macro `SCOPESTATICS` and two unused forward declarations.

```suggestion
```

Silence-112 committed:
Update A3A/addons/ultimate/CfgVehicles.hpp

Co-authored-by: UnseenKill <post@oliver-schieche.de>
Signed-off-by: Jake Leach <78276788+Silence-112@users.noreply.github.com>

stutpip123 commented:
the biggest issue with this concept is it creates a circular dependency of loading the mod to add the objects to a map.
it'll need to exist both as an independent small mod for map creation, and also within the main mod for when its being used

Silence-112 commented:
> the biggest issue with this concept is it creates a circular dependency of loading the mod to add the objects to a map. it'll need to exist both as an independent small mod for map creation, and also within the main mod for when its being used

Behold the humble ALT + F4 and rebuild

stutpip123 commented:
> > the biggest issue with this concept is it creates a circular dependency of loading the mod to add the objects to a map. it'll need to exist both as an independent small mod for map creation, and also within the main mod for when its being used
> 
> Behold the humble ALT + F4 and rebuild

I have tried it before, it broke the editor loading the main mod in entirety

wersal454 commented:
> the biggest issue with this concept is it creates a circular dependency of loading the mod to add the objects to a map. it'll need to exist both as an independent small mod for map creation, and also within the main mod for when its being used

![huh](https://github.com/user-attachments/assets/864bfd99-6efb-4a2d-87b4-8b2c5648014b)

wersal454 commented:
> > > the biggest issue with this concept is it creates a circular dependency of loading the mod to add the objects to a map. it'll need to exist both as an independent small mod for map creation, and also within the main mod for when its being used
> > 
> > 
> > Behold the humble ALT + F4 and rebuild
> 
> I have tried it before, it broke the editor loading the main mod in entirety

before
<img width="908" height="300" alt="image" src="https://github.com/user-attachments/assets/d286e5f8-8eb4-44d3-8cfd-d943d0c87035" />
?

because, it was broken before that.

UnseenKill commented:
> > > > the biggest issue with this concept is it creates a circular dependency of loading the mod to add the objects to a map. it'll need to exist both as an independent small mod for map creation, and also within the main mod for when its being used
> > > 
> > > 
> > > Behold the humble ALT + F4 and rebuild
> > 
> > 
> > I have tried it before, it broke the editor loading the main mod in entirety
> 
> before <img alt="image" width="908" height="300" src="https://private-user-images.githubusercontent.com/29953387/557539558-d286e5f8-8eb4-44d3-8cfd-d943d0c87035.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzI1NDc2ODUsIm5iZiI6MTc3MjU0NzM4NSwicGF0aCI6Ii8yOTk1MzM4Ny81NTc1Mzk1NTgtZDI4NmU1ZjgtOGViNC00NGQzLThjZmQtZDk0M2QwYzg3MDM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAzMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMzAzVDE0MTYyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3NDc3ZTIyMWMwODZjYmEyOTAzMTZlNjdjYWY3OTQ0ODU1M2IyOWU5MTY2ZTJhNjZjMmI3ZjczY2UwZjRkN2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.b8pksR2MJnP3u1iQGZETIZwmyvDro5pht_QV_rsHzg8"> ?
> 
> because, it was broken before that.

What

wersal454 commented:
> > > > > the biggest issue with this concept is it creates a circular dependency of loading the mod to add the objects to a map. it'll need to exist both as an independent small mod for map creation, and also within the main mod for when its being used
> > > > 
> > > > 
> > > > Behold the humble ALT + F4 and rebuild
> > > 
> > > 
> > > I have tried it before, it broke the editor loading the main mod in entirety
> > 
> > 
> > before <img alt="image" width="908" height="300" src="https://private-user-images.githubusercontent.com/29953387/557539558-d286e5f8-8eb4-44d3-8cfd-d943d0c87035.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzI1NDc2ODUsIm5iZiI6MTc3MjU0NzM4NSwicGF0aCI6Ii8yOTk1MzM4Ny81NTc1Mzk1NTgtZDI4NmU1ZjgtOGViNC00NGQzLThjZmQtZDk0M2QwYzg3MDM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAzMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMzAzVDE0MTYyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3NDc3ZTIyMWMwODZjYmEyOTAzMTZlNjdjYWY3OTQ0ODU1M2IyOWU5MTY2ZTJhNjZjMmI3ZjczY2UwZjRkN2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.b8pksR2MJnP3u1iQGZETIZwmyvDro5pht_QV_rsHzg8"> ?
> > because, it was broken before that.
> 
> What

that's what I'm asking

Silence-112 commented:
![yeet](https://github.com/user-attachments/assets/6de9041b-2c37-42cc-9bfd-0ff7b0a376e3)
